projects/core/src/checkout/store/reducers/payment-types.reducer.ts

```diff
@@ -36,6 +38,27 @@ export function reducer(
       };
     }
 
+    case CheckoutActions.SET_PAYMENT_TYPE: {
+      return {
+        ...state,
+        selected: action.payload.typeCode,
+      };
+    }
```

Comment:
Don't we usually use the `_SUCCESS` action to update the state?

Reply:
Yes, this is done in #7577.